### PR TITLE
fix: correct scoring algorithm, add error boundaries, code splitting, and production routing

### PR DESCRIPTION
**Pull request by luandro**
## Summary

Five targeted fixes that address broken recommendation scoring, production deployment routing, translation crash risk, unhandled render errors, and initial bundle size:

1. **Correct scoring algorithm priority-to-attribute mapping** (`RecommendationResults.tsx`) — The `pointWeights` mapping was completely inverted: `easyToUse` mapped to `energy`, `lowPower` to `concurrency`, `lowCost` to `formatEase` AND `cost`, etc. Now correctly maps `easyToUse→formatEase`, `lowPower→energy`, `scalable→growth`, `lowCost→cost`. The `language` priority (no corresponding hardware attribute) is redistributed proportionally among the other 4 priority-driven attributes. Concurrency is derived from the `users` answer instead of priority points. Match percentage now uses actual weight sum instead of hardcoded 5, with a division-by-zero guard.

2. **Guard `t()` with `returnObjects: true` against non-array returns** (`i18n-helpers.ts`, `PDFTemplate.tsx`, `RecommendationResults.tsx`) — New `tArray()` helper wraps `t(key, { returnObjects: true })` with `Array.isArray()` check, returning `[]` for missing keys. Replaces 6 unsafe `.map()` calls that would iterate over individual characters when a translation key is missing.

3. **Fix production router basename** (`.env.production`) — Adds `VITE_ROUTER_BASENAME=/community-box/` so React Router matches routes from the correct base path in production, fixing direct URL access on GitHub Pages.

4. **Add ErrorBoundary** (`ErrorBoundary.tsx`, `App.tsx`, `Questionnaire.tsx`) — Class component error boundary wrapping the route tree (catches fatal errors) and a dedicated boundary around `RecommendationResults` (protects questionnaire progress on render errors). Supports `onReset`, `resetLabel`, and custom `fallback` props. Uses non-destructive retry ("Try Again") for chunk-load failures.

5. **Implement route-level code splitting with retryable lazy loading** (`App.tsx`) — `LandingPage`, `Index`, `DocsHome`, and `MarkdownPage` converted to lazy imports via `retryableLazy` helper. This helper works around React.lazy's cached rejected imports by creating fresh `lazy()` wrappers on retry, falling back to `window.location.reload()` after 2 failed attempts. Suspense fallback shows a loading spinner. Build now produces separate chunks per route.

## Context

Phase 1 of the low-hanging fruit improvements plan. These are the highest-priority fixes addressing broken recommendations, production deployment issues, crash risks, and bundle size.

## Files Changed

| File | Change |
|------|--------|
| `.gitignore` | Added `app/plans/` and agent tool directories |
| `app/.env.production` | Added `VITE_ROUTER_BASENAME=/community-box/` |
| `app/src/App.tsx` | `retryableLazy` helper, `LoadingSpinner`, `AppRoutes` component, `ErrorBoundary` wrapping, `Suspense` |
| `app/src/components/ErrorBoundary.tsx` | New class component with `getDerivedStateFromError`, `componentDidCatch`, configurable reset |
| `app/src/components/PDFTemplate.tsx` | Replaced 2 unsafe `t()` calls with `tArray()` |
| `app/src/components/Questionnaire.tsx` | Wrapped `RecommendationResults` in dedicated `ErrorBoundary` |
| `app/src/components/RecommendationResults.tsx` | Fixed `pointWeights` mapping, language redistribution, concurrency derivation, match % calculation, guarded `answers.users`, replaced 4 unsafe `t()` calls with `tArray()` |
| `app/src/context/QuestionnaireContext.tsx` | Added `skipUrlUpdateRef` to fix `resetSurvey` URL race condition |
| `app/src/lib/i18n-helpers.ts` | New `tArray()` helper for safe `returnObjects` usage |

## Validation

- `npm run lint` — passes cleanly
- `npm run build` — succeeds, produces separate route chunks (LandingPage, DocsHome, MarkdownPage, Index)
- All 12 review threads resolved
- No pending changes requested or approvals blocking

> [!NOTE]
> ### Fix device scoring algorithm and add error boundaries, code splitting, and production routing
> - Centralizes device recommendation logic into [deviceScoring.tsx](https://github.com/coolabnet/community-box/pull/39/files#diff-75b9412cd8019b0d34128b0db0085183cda4497267af859a40fc92238bae92bf), fixing normalization mappings and introducing language-point redistribution and concurrency baseline weighting that changes how devices are ranked and scored
> - Adds a reusable [ErrorBoundary](https://github.com/coolabnet/community-box/pull/39/files#diff-cd7d4545b8fdc6296f8b73f16ed459e70bae27b6833e737ac13c6fa5edc4d5b9) component and wraps lazy-loaded routes and `RecommendationResults` with it, providing reset and reload options on failure
> - Splits route components into lazy-loaded chunks via [retryableLazy](https://github.com/coolabnet/community-box/pull/39/files#diff-73b923a7a865109cb391593531ad582df15e836bd976e26a398454bea16f6214), with a `LoadingSpinner` fallback and special handling for chunk-load errors
> - Sets `VITE_ROUTER_BASENAME=/community-box/` in [.env.production](https://github.com/coolabnet/community-box/pull/39/files#diff-8a83c34bfbd7090bd6f25625ae1e6c5947792645818de73424fd85672c927ebd) for correct routing under the production subpath
> - Adds a `tArray` i18n helper to safely return translation arrays, fixing character-by-character iteration bugs in `PDFTemplate` and `RecommendationResults` when keys are missing
> - Risk: scoring changes (normalization scales, weight redistribution) will alter device recommendation results compared to the previous inline algorithm
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 8f650c7.</sup>
> 
>

### CODE REVIEW

File: app/src/components/ErrorBoundary.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  static getDerivedStateFromError(error: Error): ErrorBoundaryState {
+    return { hasError: true, error };
+  }
+
+  componentDidCatch(error: Error, errorInfo: ErrorInfo): void {
+    console.error('ErrorBoundary caught an error:', error, errorInfo);
+  }
+
+  resetErrorBoundary = (): void => {
```
[the remark this answers is not in the document]
Comment (luandro) [reply app/src/components/ErrorBoundary.tsx:39]: Fixed in 9162a4f: Added `onReset` callback prop to ErrorBoundary so questionnaire state can be cleared on retry. Added "Reload Page" button for lazy-load chunk failures. Updated copy to be contextually accurate ("Start Over" when onReset is provided, "Try Again" otherwise). Passed `onReset={handleStartOver}` in Questionnaire.tsx. Also updated `fallback` prop to accept a render function receiving `resetErrorBoundary`.

File: app/plans/2026-04-19-2026-04-18-low-hanging-fruit-improvements-4.0.md [hunk 2]
```diff
@@ -0,0 +1,627 @@
...
+> **Status Legend**: `- [ ]` Not Started | `- [~]` In Progress | `- [x]` Completed | `- [-]` Cancelled
+
+## Objective
+
+Systematically address identified low-hanging fruit improvements across the Community Box App codebase, prioritized by impact and effort. All product decisions have been resolved — the plan is ready for implementation.
+
+### Decisions Resolved
+
+- **Dark mode (Task 7)**: Ship it. Add `ThemeProvider` and toggle button. Infrastructure is already 80% in place.
+- **Language priority mapping (Task 1)**: Exclude `language` from device scoring. Language support depends on OS/software, not hardware. Points allocated to "Works in your language" will be redistributed proportionally among the other 4 device-scoring attributes. Language will continue to influence OS/service recommendations via `getOSSuggestions`.
```
[the remark this answers is not in the document]
Comment (luandro) [reply app/plans/2026-04-19-2026-04-18-low-hanging-fruit-improvements-4.0.md]: Fixed in 9162a4f: Updated plan to accurately describe scoring behavior. Language points are now redistributed proportionally among energy/growth/formatEase/cost (equal split if all zero). Corrected false claim about language influencing OS/service recommendations — `getOSSuggestions` and `getRecommendations` do not receive language input. This may be addressed in a future task.

File: app/src/components/RecommendationResults.tsx [hunk 3]
```diff
@@ -176,12 +177,15 @@ const calculateDeviceScores = (
...
+  // language is excluded (no corresponding hardware attribute; weight redistributes via normalization)
+  // concurrency and reusable are not driven by priorities (driven by `users` answer and `reuse` special case)
   const pointWeights: Record<AttributeKey, number> = {
-    energy: points.easyToUse ?? 0,
-    concurrency: points.lowPower ?? 0,
-    growth: points.language ?? 0,
-    reusable: points.scalable ?? 0,
-    formatEase: points.lowCost ?? 0,
+    energy: points.lowPower ?? 0,
+    concurrency: 0,
```
[the remark this answers is not in the document]
Comment (luandro) [reply app/src/components/RecommendationResults.tsx]: Fixed in 9162a4f: `concurrency` weight is now derived from the `users` answer (1-2→0.5, 3-5→1.5, 6+→3) instead of being hardcoded to 0. Language points are redistributed proportionally among the 4 priority-driven attributes. Both fixes ensure multi-user needs and language allocations correctly influence device rankings.

File: app/src/App.tsx [hunk 4]
```diff
@@ -22,19 +31,22 @@ const App = () => {
...
-        </BrowserRouter>
+        <ErrorBoundary>
+          <BrowserRouter basename={basename}>
+            <Suspense fallback={<LoadingSpinner />}>
+              <Routes>
+                <Route path="/" element={<LandingPage />} />
+                <Route path="/questionnaire" element={<Index />} />
+                <Route path="/docs" element={<DocsLayout />}>
+                  <Route index element={<DocsHome />} />
+                  <Route path="*" element={<ErrorBoundary><MarkdownPage /></ErrorBoundary>} />
```
[the remark this answers is not in the document]
Comment (luandro) [reply app/src/App.tsx]: Fixed: extracted AppRoutes component using useLocation() and added key={location.pathname} to ErrorBoundary on docs wildcard route, forcing remount on slug change.

File: app/plans/2026-04-19-2026-04-18-low-hanging-fruit-improvements-4.0.md [hunk 5]
```diff
@@ -0,0 +1,627 @@
...
+**File**: `src/context/QuestionnaireContext.tsx:45-99`
+
+**Problem**: The context value object is recreated on every render. Functions `setAnswer`, `goBack`, and `resetSurvey` are recreated on every render, causing all consumers (`Questionnaire`, `QuestionCard`, `PointsAllocation`, `UsageSelection`) to re-render unnecessarily.
+
+Additionally, `resetSurvey()` has a race condition: it clears the URL via `window.history.replaceState({}, '', window.location.pathname)` at line 74, but then `setCurrentStep(0)` and `setAnswers({})` trigger the `useEffect` on line 51 which calls `updateUrlWithState(state)` — re-encoding `?step=0` into the URL immediately.
+
+- [ ] Wrap `setAnswer` with `useCallback`
+- [ ] Wrap `goBack` with `useCallback`
+- [ ] Wrap `resetSurvey` with `useCallback`
+- [ ] Fix `resetSurvey` URL race condition: remove the manual `window.history.replaceState` call and instead let the `useEffect` handle URL clearing naturally (since `setCurrentStep(0)` + `setAnswers({})` will trigger it). Alternatively, add a `skipUrlUpdate` ref that the effect checks.
```
[the remark this answers is not in the document]
Comment (luandro) [reply app/plans/2026-04-19-2026-04-18-low-hanging-fruit-improvements-4.0.md]: Fixed: corrected the plan file recommendation and added skipUrlUpdateRef to QuestionnaireContext.tsx to prevent the useEffect from re-encoding ?step=0 after resetSurvey().

File: app/src/components/RecommendationResults.tsx [hunk 6]
```diff
@@ -176,15 +177,52 @@ const calculateDeviceScores = (
...
+  }
+
+  // Derive concurrency baseline weight from the `users` answer
+  // More users → higher concurrency weight so multi-user needs influence device ranking
+  const concurrencyMap: Record<string, number> = {
+    '1-2': 0.5,   // Low concurrency need — small baseline weight
+    '3-5': 1.5,   // Moderate — meaningful weight
+    '6+': 3,      // High — strong influence on ranking
+  };
+  pointWeights.concurrency = concurrencyMap[answers.users] ?? 1;
```
[the remark this answers is not in the document]
Comment (luandro) [reply app/src/components/RecommendationResults.tsx]: Fixed: concurrency weight now only applied when user allocated priority points (priorityKeys.some(key => pointWeights[key] > 0)), preserving the equal-weight fallback when no points are spent.

File: app/src/App.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (luandro) [reply app/src/App.tsx]: Fixed: extracted AppRoutes component with useLocation() and added key prop to ErrorBoundary on docs wildcard route.

File: app/plans/2026-04-19-2026-04-18-low-hanging-fruit-improvements-4.0.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (luandro) [reply app/plans/2026-04-19-2026-04-18-low-hanging-fruit-improvements-4.0.md]: Fixed: corrected plan file recommendation and added skipUrlUpdateRef to QuestionnaireContext.tsx to prevent useEffect re-encoding step=0 after resetSurvey().

File: app/src/components/RecommendationResults.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (luandro) [reply app/src/components/RecommendationResults.tsx]: Fixed: concurrency weight now only applied when user allocated priority points, preserving equal-weight fallback when no points spent.

**Comment by luandro** [comment]
Addressed all 3 review threads:

1. **ErrorBoundary stale state on docs navigation** (`App.tsx`): Extracted `AppRoutes` inner component using `useLocation()` and added `key={location.pathname}` to the ErrorBoundary on the docs wildcard route, forcing a remount when navigating between docs pages.

2. **resetSurvey URL race condition** (`QuestionnaireContext.tsx` + plan file): Corrected the plan file's incorrect recommendation. Added `skipUrlUpdateRef` to prevent the useEffect from re-encoding `?step=0` after `resetSurvey()` clears the URL.

3. **Concurrency weight breaks equal-weight fallback** (`RecommendationResults.tsx`): Concurrency weight is now only applied when the user actually allocated priority points (`priorityKeys.some(key => pointWeights[key] > 0)`), preserving the equal-weight fallback when no points are spent.

All changes pass lint. Code reviewed for correctness with no issues found.

File: app/src/App.tsx [hunk 7]
```diff
@@ -1,18 +1,27 @@
...
 import { BrowserRouter, Routes, Route } from "react-router-dom";
-import Index from "./pages/Index";
 import NotFound from "./pages/NotFound";
-import LandingPage from "./pages/LandingPage";
 import DocsLayout from "./pages/DocsLayout";
-import DocsHome from "./pages/DocsHome";
-import MarkdownPage from "./pages/MarkdownPage";
+import ErrorBoundary from "@/components/ErrorBoundary";
+
+const LandingPage = lazy(() => import("./pages/LandingPage"));
```
[the remark this answers is not in the document]
Comment (luandro) [reply app/src/App.tsx]: Fixed in 6bb8082: wrapped all lazy imports in retryableLazy helper that uses useState + useMemo to create a fresh lazy() on ErrorBoundary reset, bypassing React.lazy's cached rejection for chunk-load failures.

File: app/src/App.tsx [hunk 8]
```diff
@@ -1,18 +1,72 @@
...
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- count is intentionally used to trigger lazy() recreation on retry
+    const LazyComp = useMemo(() => lazy(factory), [count]);
+    const handleReset = () => {
+      retryCount++;
+      setCount(retryCount);
+    };
+    return (
+      <ErrorBoundary onReset={handleReset}>
+        <LazyComp {...(props as Record<string, unknown>)} />
+      </ErrorBoundary>
```
[the remark this answers is not in the document]
Comment (luandro) [reply app/src/App.tsx]: Fixed in cd79eb6: Added resetLabel prop to ErrorBoundary. retryableLazy now passes resetLabel="Try Again" so chunk-load retries show the correct label instead of misleading "Start Over".

File: app/src/context/QuestionnaireContext.tsx [hunk 9]
```diff
@@ -47,8 +47,14 @@ export function QuestionnaireProvider({ children }: { children: React.ReactNode
...
   const totalSteps = 10; // 6 questions + points + usage + main use + results
 
+  const skipUrlUpdateRef = useRef(false);
+
   // Save state to localStorage and update URL whenever state changes
   useEffect(() => {
+    if (skipUrlUpdateRef.current) {
+      skipUrlUpdateRef.current = false;
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (luandro) [reply app/src/context/QuestionnaireContext.tsx:58]: Fixed in cd79eb6: Added setTimeout(0) safety net in resetSurvey() to ensure skipUrlUpdateRef is always cleared, even when React bails out of re-rendering because state is already at initial values.

File: app/src/components/Questionnaire.tsx [hunk 10]
```diff
@@ -153,11 +154,13 @@ const Questionnaire = () => {
       );
     } else if (currentQuestion.id === 'results') {
       return (
-        <RecommendationResults
-          key="recommendation-results"
-          answers={answers as UserAnswers}
-          onStartOver={handleStartOver}
-        />
+        <ErrorBoundary onReset={handleStartOver}>
```
[the remark this answers is not in the document]
Comment (luandro) [reply app/src/components/Questionnaire.tsx]: Fixed in ea5adbb: ErrorBoundary around RecommendationResults now uses non-destructive retry (resetLabel="Try Again") instead of calling resetSurvey(). The separate 'Start Over' button inside RecommendationResults remains for intentional resets.

File: app/src/App.tsx [hunk 11]
```diff
@@ -1,18 +1,72 @@
...
+ * the import.
+ */
+const retryableLazy = <T extends ComponentType<unknown>>(
+  factory: () => Promise<{ default: T }>
+) => {
+  let retryCount = 0;
+  const Component = (props: React.ComponentProps<T>) => {
+    const [count, setCount] = useState(retryCount);
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- count is intentionally used to trigger lazy() recreation on retry
+    const LazyComp = useMemo(() => lazy(factory), [count]);
```
[the remark this answers is not in the document]
Comment (luandro) [reply app/src/App.tsx]: Fixed in ea5adbb: retryableLazy now falls back to window.location.reload() after MAX_RETRIES (2) failed attempts, bypassing browser-cached failed module imports by fetching fresh HTML with fresh chunks.

File: app/src/components/RecommendationResults.tsx [hunk 12]
```diff
@@ -176,15 +177,52 @@ const calculateDeviceScores = (
...
+  // More users → higher concurrency weight so multi-user needs influence device ranking
+  const concurrencyMap: Record<string, number> = {
+    '1-2': 0.5,   // Low concurrency need — small baseline weight
+    '3-5': 1.5,   // Moderate — meaningful weight
+    '6+': 3,      // High — strong influence on ranking
+  };
+  pointWeights.concurrency = priorityKeys.some(key => pointWeights[key] > 0) ? (concurrencyMap[answers.users] ?? 1) : 0;
+
   // Calculate total points allocated
   const totalPoints = Object.values(pointWeights).reduce((sum, points) => sum + points, 0);
```
[the remark this answers is not in the document]
Comment (luandro) [reply app/src/components/RecommendationResults.tsx]: Fixed in ea5adbb: Guarded answers.users in both normalizeUserAnswers and calculateDeviceScores with fallback to '1-2' (single-user baseline). Map lookup now falls back to 0.5 instead of 1, preventing inflated concurrency scores.

File: app/src/components/Questionnaire.tsx [hunk 13]
```diff
@@ -153,11 +156,13 @@ const Questionnaire = () => {
       );
     } else if (currentQuestion.id === 'results') {
       return (
-        <RecommendationResults
-          key="recommendation-results"
-          answers={answers as UserAnswers}
-          onStartOver={handleStartOver}
-        />
+        <ErrorBoundary resetLabel="Try Again">
```
[the remark this answers is not in the document]
Comment (luandro) [reply app/src/components/Questionnaire.tsx]: Fixed in d5951cd: replaced hardcoded 'Try Again' with t('errors.tryAgain') and added translation keys to en/pt/es locales.

File: app/src/components/RecommendationResults.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (luandro) [reply app/src/components/RecommendationResults.tsx]: Already guarded: `answers.users ?? '1-2'` at line 171 (`normalizeUserAnswers`) and line 231 (`calculateDeviceScores`), with additional fallbacks `|| 3` and `?? 0.5` for unexpected values. No change needed.

File: app/src/App.tsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (luandro) [reply app/src/App.tsx]: Already handled: `retryableLazy` detects chunk-load errors via `isChunkLoadError()` and shows only a "Reload Page" button (calling `window.location.reload()`) instead of retrying the cached import. The ErrorBoundary retry path also forces a full page reload via `onRetry={() => window.location.reload()}`. No change needed.

File: app/src/components/RecommendationResults.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (luandro) [reply app/src/components/RecommendationResults.tsx]: Already guarded: answers.users is protected with nullish coalescing (?? '1-2') at line 171 (normalizeUserAnswers) and line 231 (calculateDeviceScores), with additional fallbacks (|| 3 and ?? 0.5) for unexpected values. No change needed.

File: app/src/App.tsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (luandro) [reply app/src/App.tsx]: Already handled: retryableLazy detects chunk-load errors via isChunkLoadError() and shows only a Reload Page button (calling window.location.reload()) instead of retrying the cached import. The ErrorBoundary retry path also forces a full page reload via onRetry. No change needed.

**Comment by luandro** [comment]
## QA Checklist

### 1. Scoring Algorithm (commits `61b77cc`, `adb9b81`, `42a9418`, `e1163d6`)

**Screen**: Questionnaire results page (step 10)

| What to check | How | Expected |
|---|---|---|
| Energy scoring is correct | Answer "Quedas frequentes" (frequent outages) to electricity | Raspberry Pi or Reused PC ranks high (energy-efficient devices favored) |
| Concurrency weight from `users` answer | Answer "6+" users | Intel NUC or Reused PC ranks higher (high concurrency devices) |
| Reuse special case preserved | Answer "Sim" (yes) to reuse question | Reused PC gets a significant boost (40% weight to `reusable` attribute) |
| Point redistribution works | Allocate all 5 points to "language" priority | Points redistributed equally among energy/growth/formatEase/cost (no crash, no zero-division) |

### 2. ErrorBoundary and Lazy Loading (commits `ec735be`, `24614f7`, `036559d`, `b878783`, `d5951cd`, `8f650c7`)

**Screen**: Error fallback UI

| What to check | How | Expected |
|---|---|---|
| Chunk-load error shows single "Reload Page" | DevTools > Network > block any `.js` chunk > navigate to trigger it | Single "Reload Page" button, no duplicate buttons |
| Non-chunk runtime error shows distinct buttons | Throw an error inside a lazy component (add `throw new Error("test")` to its render) | Two buttons: primary **"Retry Load"** + secondary **"Reload Page"** |
| Results ErrorBoundary "Try Again" resets state | Complete questionnaire, then add `throw new Error("crash")` to `RecommendationResults.tsx:93` | Clicking "Tentar novamente" resets to step 1 with cleared answers (no crash loop) |
| ErrorBoundary text is translated | Switch language to English, then trigger an error | Button shows "Try Again" in English, not hardcoded Portuguese |

### 3. Code Splitting (commits `61ef4cf`, `24614f7`)

**Screen**: Network tab during navigation

| What to check | How | Expected |
|---|---|---|
| Lazy chunks load on navigation | Open DevTools Network tab, navigate between pages | Separate `.js` chunks load per route (not one monolithic bundle) |
| Loading spinner shows during chunk load | Throttle network to "Slow 3G" in DevTools, navigate | Spinning circle appears while chunk loads |
| Route-based code splitting works | Navigate to `/questionnaire`, `/docs`, etc. | Each route loads independently, no full-page reload |

### 4. Dynamic `totalSteps` (commit `8f650c7`)

**Screen**: Questionnaire progress bar

| What to check | How | Expected |
|---|---|---|
| Progress bar shows correct fraction | Start questionnaire, look at progress bar on step 1 | Shows ~10% (1 of 10 steps) |
| Progress advances correctly | Complete each step | 10% → 20% → ... → 100% |
| Progress bar hides on results | Reach step 10 (results) | No progress bar visible |
| `totalSteps` derives from questions array | Temporarily comment out one question in `Questionnaire.tsx:18-104`, reload | Progress shows 1 of 9 (~11%), not 1 of 10 |

### 5. Fast Refresh (commit `8f650c7`)

**Screen**: Dev server terminal + browser console

| What to check | How | Expected |
|---|---|---|
| No Fast Refresh warnings | Run `npm run dev`, edit `RecommendationResults.tsx` (add a comment), save | Terminal shows clean HMR update, no `react-refresh/only-export-components` warning |
| Hot reload works | Same test as above | Component updates in browser without full page refresh |
| Scoring functions in separate module | Check `src/utils/deviceScoring.tsx` | Contains `normalizeUserAnswers`, `calculateDeviceScores`, `devices` |

### 6. Router Base Path (commits `96eb24c`, `c7c0e77`)

**Screen**: Deployed GitHub Pages site

| What to check | How | Expected |
|---|---|---|
| Routes work on GitHub Pages | Visit deployed URL (e.g. `username.github.io/community-box/questionnaire`) | Page loads correctly, no 404 |
| Assets load with correct base path | Open deployed site, check Network tab | All `.js`, `.css`, images load from `/community-box/...` paths |

### 7. Automated Validation (run before merge)

```bash
npm run lint && npx vitest run && npm run build
```

All three must pass with zero errors. The test suite has 72 tests covering scoring edge cases.

---

### Quick Smoke Test (5 minutes)

1. `npm run dev` and open browser
2. Click through all 10 questionnaire steps with any answers
3. Verify results page shows a device recommendation with percentage
4. Click "Ver Alternativas" and expand one alternative
5. Verify services and OS recommendations appear
6. Click "Recomeçar" — should return to step 1 with cleared state
7. Switch language to English (button in top bar) and repeat steps 2–6
8. Run `npm run lint && npx vitest run && npm run build`
